test/data/livechat/ILiveChatRepository.test.ts

```diff
@@ -0,0 +1,383 @@
+import visitorFactory from '../../factories/VisitorFactory';
+
+describe('ILiveChatRepository', () => {
+
+    let mockedCache: ILiveChatCacheDataSource = mock<ILiveChatCacheDataSource>();
+    let mockedInternal: ILiveChatInternalDataSource = mock<ILiveChatInternalDataSource>();
+    let livechatRepo: ILiveChatRepository;
+
+    describe('#getDepartmentByName()', () => {
+
+        beforeEach(() => {
+            mockedCache = mock<ILiveChatCacheDataSource>();
+            mockedInternal = mock<ILiveChatInternalDataSource>();
+            livechatRepo = new LiveChatCacheStrategyRepositoryImpl(instance(mockedCache), instance(mockedInternal));
+        });
+
+        it('should call from InternalDataSource', async () => {
+            const name = 'support';
+
+            when(mockedInternal.getDepartmentByName(name)).thenResolve(departmentFactory.build());
+
+            await livechatRepo.getDepartmentByName(name);
+
+            verify(mockedInternal.getDepartmentByName(name)).once();
+        });
+    });
+
+    describe('#eventCloseRoom()', () => {
+
+        beforeEach(() => {
+            mockedCache = mock<ILiveChatCacheDataSource>();
+            mockedInternal = mock<ILiveChatInternalDataSource>();
+            livechatRepo = new LiveChatCacheStrategyRepositoryImpl(instance(mockedCache), instance(mockedInternal));
+        });
+
+        it('should call from CacheDataSource', async () => {
+            const livechat = livechatRoomFactory.build();
+
+            when(mockedCache.deleteRoom(livechat)).thenResolve();
+
+            await livechatRepo.eventCloseRoom(livechat);
+
+            verify(mockedCache.deleteRoom(livechat)).once();
+        });
+    });
+
+    describe('#sendMessage()', () => {
+
+        beforeEach(() => {
+            mockedCache = mock<ILiveChatCacheDataSource>();
+            mockedInternal = mock<ILiveChatInternalDataSource>();
+            livechatRepo = new LiveChatCacheStrategyRepositoryImpl(instance(mockedCache), instance(mockedInternal));
+        });
+
+        it('should call from InternalDataSource', async () => {
+            const text = 'text from message';
+            const attachments = mock<Array<IMessageAttachment>>();
+            const livechat: ILivechatRoom = livechatRoomFactory.build();
+
+            when(mockedInternal.sendMessage(text, attachments, livechat)).thenResolve('2hSb3rKy8fn5uwWd');
+
+            await livechatRepo.sendMessage(text, attachments, livechat);
+
+            verify(mockedInternal.sendMessage(text, attachments, livechat)).once();
+        });
+    });
+
+    describe('#getRoomByVisitorToken()', () => {
+
+        beforeEach(() => {
+            mockedCache = mock<ILiveChatCacheDataSource>();
+            mockedInternal = mock<ILiveChatInternalDataSource>();
+            livechatRepo = new LiveChatCacheStrategyRepositoryImpl(instance(mockedCache), instance(mockedInternal));
+        });
+
+        it('should call from CacheDataSource', async () => {
```

Comment:
Test unnecessary.

Reply:
Or you can rename this test to "should return a visitor when the given token is valid" and check the return (equal).